sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -146,6 +147,18 @@ object RapidsPluginUtils extends Logging {
     }
     // set driver timezone
     conf.set(RapidsConf.DRIVER_TIMEZONE.key, ZoneId.systemDefault().normalized().toString)
+
+    // If spark.rapids.sql.multiThreadedRead.numThreads is not set explicitly, or it is set
+    // explicitly as value 0, then we derive it from other settings. Otherwise, we keep the
+    // users' setting.
+    val numThreadsKey = RapidsConf.MULTITHREAD_READ_NUM_THREADS.key
+    if (!conf.contains(numThreadsKey) || conf.get(numThreadsKey).toInt == 0) {
+      // Derive it from spark.executor.cores, since spark.executor.cores is not set on all cluster
+      // managers by default, we should judge whether if it's set explicitly.
+      if (conf.contains(EXECUTOR_CORES_KEY)) {
+        conf.set(numThreadsKey, Math.max(20, conf.get(EXECUTOR_CORES_KEY).toInt).toString)
```

Comment:
it might be nice to log a messages saying what we are setting this to

Reply:
it would also be nice to make a global variable (maybe something like MULTITHREAD_READ_NUM_THREADS_DEFAULT) for the 20 so that we don't have different values.